doc: nRF9160: Update Getting started to use Cellular Monitor (#12294)
nrfconnect/sdk-nrf: doc/nrf/device_guides/working_with_nrf/nrf91/nrf9160.rst

      :alt: Programmer - Add file

      Programmer - Add file

#. Navigate to where you extracted the firmware, and then to the :file:`img_app_bl` folder there.
#. Select the :file:`.hex` file for the application you are programming.

   For Asset Tracker v2, select the :file:`nrf9160dk_asset_tracker_v2_<version-number>.hex` HEX file.

   For NB-IoT, there is a second variant of the Asset Tracker v2 firmware in the :file:`nrf9160dk_asset_tracker_v2_nbiot_legacy_pco_<version-number>.hex` file.
   Only use this legacy variant if your network does not support ePCO.

#. Click the :guilabel:`Erase & write` button in the **DEVICE** section to program the DK.
   Do not unplug or turn off the DK during this process.

   .. figure:: images/programmer_erasewrite_nrf9160dk.png
      :alt: Programmer - Erase & write

      Programmer - Erase & write

.. _build_pgm_nrf9160:

Building and programming
************************

You can program applications and samples on the nRF9160 DK after obtaining the corresponding firmware images.

.. note::
   When you update the application firmware on an nRF9160 DK, make sure the modem firmware and application firmware are compatible versions.

> To program applications using the Programmer app from `nRF Connect for Desktop`_, follow the instrutions in :ref:`nrf9160_updating_fw_application`.

Reviewer: May be we could remove this sentence because the section is just above this section.

Download the latest application and modem firmware from the `nRF9160 DK Downloads`_ page.

The downloaded ZIP archive contains the following firmware:

Application firmware
  The :file:`img_app_bl` folder contains full firmware images for different applications.
  The guides in this section use the image for the :ref:`asset_tracker_v2` application as an example.
  Asset Tracker v2 simulates sensor data and transmits it to Nordic Semiconductor's cloud solution, `nRF Cloud`_.

  The data is transmitted using either LTE-M or NB-IoT.
  Check with your SIM card provider for the mode they support at your location.
  For the iBasis SIM card provided with the nRF9160 DK, see `iBasis IoT network coverage`_.

>   For locations where both are supported, Asset Tracker v2 first attempts to use LTE-M, then NB-IoT.

Author: Excellent point. I'll also move this sentence a bit earlier in the paragraph..

If you experience any problems during the process, restart the Programmer app by pressing ``Ctrl+R`` (``command+R`` in macOS), and try again.

1. Open the Programmer app.
#. Make sure the **PROG/DEBUG SW10** switch (**SW5** on DK v0.9.0 and earlier) on the nRF9160 DK is set to **nRF91** or **nRF52** as appropriate for the application or sample you are programming.
   See the `Device programming section in the nRF9160 DK User Guide`_ for more information.

   For the :ref:`asset_tracker_v2` application, it must be set to **nRF91**.

#. Connect the nRF9160 DK to the computer with a micro-USB cable, and then turn the DK on.
#. Click :guilabel:`SELECT DEVICE` and select the DK from the drop-down list.
   You can identify the nRF9160 DK by the fact that it has three COM ports when you expand its entry.

   .. figure:: images/programmer_com_ports.png
      :alt: Programmer - COM ports

      Programmer - COM ports

   If the three COM ports are not visible, press ``Ctrl+R`` in Windows or ``command+R`` in macOS to restart the Programmer application.

   The drop-down text changes to the type of the selected device, with its SEGGER ID below the name.

>    The Device Memory Layout section also changes its name to the device name, and indicates that the device is connected.

Reviewer 2: ```suggestion
   The **Device memory layout** section also changes its name to the device name, and indicates that the device is connected.
```

   If the :guilabel:`Auto read memory` option is selected in the **DEVICE** section, the memory layout will update.
   If it is not selected and you wish to see the memory layout, click :guilabel:`Read` in the **DEVICE** section.

#. Click :guilabel:`Add file` in the **FILE** section, and select :guilabel:`Browse`.

   .. figure:: images/programmer_addfile_nrf9160dk.png
      :alt: Programmer - Add file

      Programmer - Add file

#. Navigate to where you extracted the firmware, and then to the :file:`img_app_bl` folder there.
#. Choose the :file:`.hex` file for the application you are programming.

   For Asset Tracker v2, select the :file:`nrf9160dk_asset_tracker_v2_<version-number>.hex` HEX file.

>    For NB-IoT, there is a second variant of the Asse Tracker v2 firmware in the :file:`nrf9160dk_asset_tracker_v2_nbiot_legacy_pco_<version-number>.hex` file.

Reviewer 2: ```suggestion
   For NB-IoT, there is a second variant of the Asset Tracker v2 firmware in the :file:`nrf9160dk_asset_tracker_v2_nbiot_legacy_pco_<version-number>.hex` file.
```
second or another?

      The device resets and runs the programmed sample or application.

.. _nrf9160_ug_updating_fw_programmer:

> Updating the DK firmware using Programmer

Author: I was kinda 50-50 on that myself, so I'll go ahead and make that change.

      The device resets and runs the programmed sample or application.

.. _nrf9160_ug_updating_fw_programmer:

Updating the DK firmware using Programmer
=========================================

Complete the following steps to program applications using the Programmer app from `nRF Connect for Desktop`_.

.. _nrf9160_updating_fw_modem:
.. _nrf9160_gs_updating_fw_modem:

Updating the modem firmware
---------------------------

To update the modem firmware, complete the following steps.
If you experience any problems during the process, press ``Ctrl+R`` (``command+R`` on macOS) to restart the Programmer app and try again.

.. note::

>    Updating the modem firmware erases the contents of the flash memory, so you must program the application again to the nRF9160 DK.

Author: Hmm, I'm guessing this must be outdated information. It's been preserved across several versions of NCS. Would replacing the note with something about using matching modem firmware and application firmware be warranted?

>    Updating the modem firmware erases the contents of the flash memory, so you must program the application again to the nRF9160 DK.

Reviewer 3: Sure, it might be good to mention that. Just wondering if this is the right place though, but that's something you can decide.

>    Updating the modem firmware erases the contents of the flash memory, so you must program the application again to the nRF9160 DK.

Author: Added just a single note in the section that describes what is in the firmware download.

1. Open the Programmer app.
#. Make sure the **PROG/DEBUG SW10** switch on the nRF9160 DK is set to **nRF91**.
   On DK v0.9.0 and earlier, this is the **SW5** switch.
#. Connect the nRF9160 DK to the computer with a micro-USB cable, and then turn the DK on.
#. Click :guilabel:`SELECT DEVICE` and select the DK from the drop-down list.

>    You can identify the nRF9160 DK by the fact that it has three COM ports.

Reviewer 3: To see the COM ports, you have to hover over the button and click the small down arrow. I didn't even know that you could do that.

Isn't it easier to identify the DK by the fact that it says "nRF9160 DK"? 😄 Maybe this whole sentence could be just removed.

![image](https://github.com/nrfconnect/sdk-nrf/assets/2588548/fdea4b38-0773-4acf-aa59-6e7dff9306c7)

Reviewer 3: That's exactly what I was thinking. It's probably better to use screenshots with the default view, because not all people (like me) even know that the device information can be expanded. 😄

   .. figure:: images/programmer_com_ports.png
      :alt: Programmer - COM ports

      Programmer - COM ports

>    If the three COM ports are not visible, press ``Ctrl+R`` in Windows or ``command+R`` in macOS to restart the Programmer application.

Reviewer 3: The COM ports are also mentioned here and in the picture above.

   The drop-down text changes to the type of the selected device, with its SEGGER ID below the name.
   The **Device memory layout** section also changes its name to the device name, and indicates that the device is connected.
   If the :guilabel:`Auto read memory` option is selected in the **DEVICE** section of the side panel, the memory layout will update.
   If it is not selected and you wish to see the memory layout, click :guilabel:`Read` in the **DEVICE** section of the side panel.

#. Click :guilabel:`Add file` in the **FILE** section, and select :guilabel:`Browse`.

   .. figure:: images/programmer_addfile_nrf9160dk.png
      :alt: Programmer - Add file

      Programmer - Add file

> #. Navigate to where you extracted the firmware, and select the :file:`mfwnrf9160_<version-number>.zip` file.

Reviewer 3: ```suggestion
#. Navigate to where you extracted the firmware, and select the :file:`mfw_nrf9160_<version-number>.zip` file.
```

.. _nrf9160_updating_fw_application:
.. _nrf9160_gs_updating_fw_application:

Updating the application firmware
---------------------------------

To update the application firmware using the Programmer app, complete the following steps.
If you experience any problems during the process, press ``Ctrl+R`` (``command+R`` in macOS) to restart the Programmer app and try again.

1. Open the Programmer app.
#. Make sure the **PROG/DEBUG SW10** switch (**SW5** on DK v0.9.0 and earlier) on the nRF9160 DK is set to **nRF91** or **nRF52** as appropriate for the application or sample you are programming.
   See the `Device programming section in the nRF9160 DK User Guide`_ for more information.

   For the :ref:`asset_tracker_v2` application, it must be set to **nRF91**.

#. Connect the nRF9160 DK to the computer with a micro-USB cable, and then turn the DK on.
#. Click :guilabel:`SELECT DEVICE` and select the DK from the drop-down list.
   You can identify the nRF9160 DK by the fact that it has three COM ports when you expand its entry.

   .. figure:: images/programmer_com_ports.png
      :alt: Programmer - COM ports

      Programmer - COM ports

>    If the three COM ports are not visible, press ``Ctrl+R`` in Windows or ``command+R`` in macOS to restart the Programmer application.

Reviewer 3: This should also be updated if you decide to remove the COM port references above.

See :ref:`cmake_options` for more details.

.. _nrf9160_gs_testing_cellular:

Testing the cellular connection with the AT Client sample
*********************************************************

The :ref:`at_client_sample` sample enables you to send AT commands to the modem of your nRF9160 DK to test and monitor the cellular connection.
You can use it to troubleshoot and debug any connection problems.

Complete the following steps to test the cellular connection using the AT Client sample:

1. Follow the steps in :ref:`nrf9160_gs_updating_fw_application` to program the sample to the DK.

>    When choosing the HEX file, choose :file:`nrf9160dk_at_client_<version-number>.hex` instead of one for Asset Tracker v2.

Reviewer 2: choose or select?
